.github/workflows/deploy-server.yml

```diff
@@ -0,0 +1,72 @@
+name: Deploy to AWS
+
+on:
+  pull_request: # for now, trigger this on PR so I can test it before merging
+  push:
+    branches:
+      - master
+
+env:
+  AWS_REGION: us-east-2
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  dockerimage:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        name: Check out code
+
+      - uses: docker/build-push-action@v1
+        name: Build and Push Docker image
+        with:
+          repository: joshprzybyszewski/cribbage
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+          tag_with_ref: true
+          add_git_labels: true
+          tag_with_sha: false
+          cache_froms: joshprzybyszewski/cribbage:master
+
+  deploy:
+    needs: dockerimage
+    name: Deploy
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+
+      - name: Configure AWS credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          aws-region: ${{ env.AWS_REGION }}
+
+      - name: Install CDK and Stack Dependencies
```

Comment:
💡 Maybe in a future improvement, we could split our frontend/backend CDK stacks and conditionally deploy them in our github actions based on the files touched in a given PR